README.md

```diff
@@ -77,46 +77,49 @@ Checks `.erb` files for syntax errors
+Note: This occurs only when the Puppet version specified is Puppet 4 or greater, or Puppet 3 with the future parser enabled.
 
 ## Configuration
 
-Paths can be excluded with:
+[TODO: where does this configuration happen? Is this in the Rakefile, or...?]
+
+* To exclude certain paths:
 
     PuppetSyntax.exclude_paths = ["vendor/**/*"]
 
-When you are using a Puppet version greater then 3.2, you can select the future parse by specifying
+* To use the "future" (Puppet 4) parser in Puppet 3.2 through 3.8, specify:
 
     PuppetSyntax.future_parser = true
 
-If you are using some form of hiera data inside your module, you can configure where the `syntax:hiera:yaml` task looks for data with:
+* To configure specific paths for the Hiera syntax check, set `hieradata_paths`. This is useful if you use Hiera data inside your module.
 
     PuppetSyntax.hieradata_paths = ["**/data/**/*.yaml", "hieradata/**/*.yaml", "hiera*.yaml"]
 
-If you are trying to validate the syntax of code written for application orchestration, you can enable the `app_management` setting:
+* To validate the syntax of code written for application orchestration, enable the `app_management` setting:
 
     PuppetSyntax.app_management = true
 
-`app_management` is supported on Puppet 4.3 or higher, defaulting to off. On Puppet 5, it is always enabled.
+  The `app_management` setting is supported with Puppet 4.3 or greater and is off by default. On Puppet 5, it is always enabled. [TODO: wait, what? Puppet 5 isn't out yet...?]
 
-Deprecation notices will cause a failure by default, you can override this functionality by setting:
+* By default, `puppet-syntax` fails on deprecation notices. [TODO: is this correct? it just said deprecation causes failure, but it didn't say what fails; I assume it's puppet-syntax] To override this behavior, set:
 
     PuppetSyntax.fail_on_deprecation_notices = false
 
-Common mistakes in key names in Hiera files will be reported:
+* To enable a syntax check on Hiera keys, set:
 
-- Leading :: in keys eg. `::notsotypical::warning2: true`
-- Single colon scope seperators eg. `:picky::warning5: true`
-- Invalid camel casing eg. `noCamelCase::warning3: true`
-- Use of hyphens eg. `no-hyphens::warning4: true`
+    PuppetSyntax.check_hiera_keys = true
 
-This can be enabled by setting
+  Common mistakes in key names in Hiera files are reported:
+
+  - Leading `::` in keys, such as: `::notsotypical::warning2: true`.
+  - Single colon scope separators, such as: `:picky::warning5: true`.
+  - Invalid camel casing, such as: `noCamelCase::warning3: true`.
+  - Use of hyphens, such as: `no-hyphens::warning4: true`.
 
-    PuppetSyntax.check_hiera_keys = true
 
 ## Installation
 
-Add this line to your application's Gemfile:
+Add this line to your application's Gemfile [TODO: Do we need to be more specific about "your application"? I'm not sure to what this refers.]
```

Comment:
I'd switch 'application' to 'module'. The original sentence is verbatim from basic new ruby project template.